### PR TITLE
feat: add venue map with draggable pin and public event map display

### LINKED ISSUE #1711: Add Venue Map View With Location Pin Correction and Show Map on Public Landing Page
The event creation wizard and common event settings already allow entering a venue address and latitude and longitude, but there is no map preview or convenient way to validate and correct the exact venue position. This leads to inaccurate locations and a weaker attendee experience. Furthermore the public landing page has neither a map and address area.

In the legacy eventyay public pages, the venue map and address are shown at the bottom of the event content area. The unified system should provide a similar experience.

Legacy reference: [https://eventyay.com/e/88882f3e](https://eventyay.com/e/88882f3e)

## Goal

* Provide a map view for the configured venue address and coordinates
* Allow organizers to correct the venue location pin visually
* Display the venue map and address on the public event landing page

Image
Image
### 1. Organizer UI: Map Preview in Event Creation and Common Settings

Where address and coordinates are configured (event creation wizard and common event settings):

* Display an embedded map using Mapbox or OpenStreetMap
* Show the current location based on:

  * Latitude and longitude if present
  * Otherwise attempt to place the pin using the address

### 2. Organizer UI: Correct Location Pin

Add an option to adjust the venue location icon:

* Allow organizers to move the pin on the map
* When the pin is moved, update the stored latitude and longitude
* Provide a clear save action and confirmation of the new coordinates

Optional but recommended:

* A reset option that re positions the pin based on the address again

### 3. Public Event Landing Page: Show Map and Address

On the public event landing page:

* Display the venue address and a map preview at the bottom of the event content area
* The map should reflect the saved latitude and longitude
* The address should be shown in readable form

This should follow the spirit of the legacy layout where the map is part of the event page content, near the bottom.

## Current Behavior

* Address and coordinates can be entered
* No embedded map is available to validate accuracy
* No way to correct pin placement via UI
* Public landing page does not show a venue map and address area

## Expected Behavior

* Organizers can preview venue location and visually validate it
* Organizers can correct the location pin by moving it
* The public landing page shows map and address at the bottom of the content area

## Acceptance Criteria

* Embedded map is available in the event creation wizard and common event settings where address and coordinates are set
* Pin is shown at the saved location
* Organizers can adjust the pin and the updated coordinates are saved
* Public landing page displays venue map and address at the bottom of the event content area
* If no venue is configured, the map section is not shown
* No regressions to event creation or settings workflows

## Notes

* Use OpenStreetMap tiles where possible to avoid unnecessary vendor lock in
* Ensure the map works well on mobile and desktop
* Consider performance and caching for public page rendering

### PR DESCRIPTION
**Pull request by prayag78**
Fixes #1711

Screenshot 2026-05-17 205621

Screenshot 2026-05-19 193547

Screenshot 2026-05-19 214930

https://github.com/user-attachments/assets/b44ad7e9-34a0-4c04-a798-18b3823ce146

https://github.com/user-attachments/assets/99ad80e6-d85b-40dc-abc6-b80e7be5a339

### CODE REVIEW

**Comment by Rachit7168** [comment]
Please address reviews

File: app/eventyay/static/leaflet/leaflet.css [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+	}
+/* .leaflet-container svg: reset svg max-width decleration shipped in Joomla! (joomla.org) 3.x */
+/* .leaflet-container img: map is broken in FF if you have max-width: 100% on tiles */
+.leaflet-container .leaflet-overlay-pane svg,
+.leaflet-container .leaflet-marker-pane img,
+.leaflet-container .leaflet-shadow-pane img,
+.leaflet-container .leaflet-tile-pane img,
+.leaflet-container img.leaflet-image-layer,
+.leaflet-container .leaflet-tile {
+	max-width: none !important;
```
Comment (Saksham-Sirohi) [inline app/eventyay/static/leaflet/leaflet.css]: avoid use of  !important

**Review by Saksham-Sirohi — CHANGES_REQUESTED** [review]
on public view replace text for venue with getting here and display address alongside on right or below the map maintaing proper gaps it should follow the pages indentation and styling

File: app/eventyay/static/leaflet/leaflet.css [hunk 2]
```diff
@@ -1,640 +1,640 @@
...
-	border-left-color: #fff;
-	}
-.leaflet-tooltip-right:before {
-	left: 0;
-	margin-left: -12px;
-	border-right-color: #fff;
-	}
+/* required styles */
+
+.leaflet-pane,
```
Comment (Rachit7168) [inline app/eventyay/static/leaflet/leaflet.css:3]: The file is changed from scss to css. Will this affect other places where this is used?

**Review by Rachit7168 — CHANGES_REQUESTED** [review]
Thanks. Please address

File: app/eventyay/static/leaflet/leaflet.css (hunk 2 above)
Comment (prayag78) [reply app/eventyay/static/leaflet/leaflet.css:3]: No, this change will not affect other places. The import change is scoped to the organizer SCSS only. The public pages continue using `leaflet.css` directly.

**Review by Saksham-Sirohi — CHANGES_REQUESTED** [review]
Image

There should be a padding below getting here, and the address text should be inside the section, starting to the right of the image with a slight nudge downwards (you can use padding top for this)

File: app/eventyay/static/pretixcontrol/scss/main.scss [hunk 3]
```diff
@@ -15,7 +15,7 @@
 @import "mails.scss";
 @import "../../pretixbase/scss/webfont.scss";
 @import "../../fileupload/jquery.fileupload.scss";
-@import "../../leaflet/leaflet.scss";
+@import "../../leaflet/leaflet.css";
```
[the remark this answers is not in the document]
Comment (prayag78) [reply app/eventyay/static/pretixcontrol/scss/main.scss]: `leaflet.scss` not exists

File: app/eventyay/static/pretixcontrol/js/ui/geo.js [hunk 4]
```diff
@@ -48,74 +79,105 @@ $(function () {
...
+
         // Map
         var $grp = $(".geodata-group", this);
-        var tiles = $grp.attr("data-tiles");
-        var attrib = $grp.attr("data-attrib");
-        if (tiles) {
+        var tiles = $grp.attr("data-tiles") || DEFAULT_TILES;
+        var attrib = $grp.attr("data-attrib") || DEFAULT_ATTRIB;
+
+        if (typeof L !== "undefined") {
```
[the remark this answers is not in the document]
Comment (prayag78) [reply app/eventyay/static/pretixcontrol/js/ui/geo.js:166]: intended behavior

File: app/eventyay/static/pretixcontrol/scss/main.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (prayag78) [reply app/eventyay/static/pretixcontrol/scss/main.scss]: keeping it as `leaflet.css` for now to avoid breaking map behavior

**Review by Rachit7168 — CHANGES_REQUESTED** [review]
I think the “Supported By / Sponsors” section would look better near the end of the event page, ideally after the venue/map section. Right now it slightly breaks the content flow between tickets and location info.

Image

**Comment by prayag78** [comment]
The issue mentions the venue map and address should be at the bottom of the event content area.

**Review by Rachit7168 — APPROVED** [review]
LGTM , Tested !

File: app/eventyay/static/pretixcontrol/js/ui/geo.js [hunk 5]
```diff
@@ -1,18 +1,26 @@
...
+        var $notifications = $(".geodata-autoupdate", this).detach();
         var $lat = $("input[name$=geo_lat]", this).first();
         var $lon = $("input[name$=geo_lon]", this).first();
         var lat;
         var lon;
-        var $updateButton = $("[data-action=update]", this);
+        var $updateButton = $notifications.find("[data-action=update]");
+        var $savePinButton = $notifications.find("[data-action=save-pin]");
+        var $resetPinButton = $notifications.find("[data-action=reset-pin]");
+        var $form = $(this).closest("form");
```
Comment (Saksham-Sirohi) [inline app/eventyay/static/pretixcontrol/js/ui/geo.js]: please avoid using this closest operator

**Review by Saksham-Sirohi — CHANGES_REQUESTED** [review]

**Review by sarafarajnasardi — APPROVED** [review]
LGTM

File: app/eventyay/presale/context.py [hunk 6]
```diff
@@ -163,6 +163,7 @@ def _default_context(request):
     except KeyError:
         ctx['html_locale'] = translation.get_language()
     ctx['settings'] = eventyay_settings
+    ctx['global_settings'] = GlobalSettingsObject().settings
     ctx['django_settings'] = settings
```
[the remark this answers is not in the document]
Comment (Saksham-Sirohi) [reply app/eventyay/presale/context.py:171]: @prayag78, why is this suppressed? Is this non-concerning?

[the remark this answers is not in the document]
Comment (prayag78) [reply app/eventyay/presale/context.py:171]: @Saksham-Sirohi I've addressed this as well in 830c5f7f453e9715a488675703d378ecf3e9ba5a exposing only the leaflet configurations

**Review by SxxAq — APPROVED** [review]
LGTM

**Review by Saksham-Sirohi — APPROVED** [review]
LGTM
